File: Main/src/masterparam.c
#include "masterparam.h"

Master     master;
RobotState robotstate;

void HandleParamInit(HandleParam* handleparam) {
  handleparam->run    = 0;
  handleparam->height = 20;
  handleparam->tilt   = 0;
  handleparam->turn   = 0;
}

void ControlParamInit(ControlParam* controlparam) {
  controlparam->begin = false;
  controlparam->stop  = true;
}

void MasterInit(Master* master) {
  HandleParamInit(&master->handle);
  ControlParamInit(&master->control);
}

void ControlParamDeal(ControlParam* controlparam) { robot.mode = controlparam->robotmode; }

void HandleParamDeal(HandleParam* handleparam) {
  // TODO: 参数暂定
  // robot.vSet						= (float) (handleparam->run * 0.01);
  // robot.L0Set					= (float) (handleparam->height * 0.01);
  // robot.rollpid.target = handleparam->tilt;
  // robot.yawpid.target	= handleparam->turn;
}

void RobotStateInit(RobotState* robotstate) {
  robotstate->pitch  = 0;  // 初始值应该是一个负值吧
  robotstate->yaw    = 0;
  robotstate->roll   = 0;
  robotstate->v      = 0;
  // TODO: 参数暂定
  robotstate->height = 0.35;
}

void RobotStateUpdate(RobotState* robotstate) {
  // TODO:
}

File: Assets/scripts/masterparam.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class masterparam
{
  //----
  // @brief 摇杆数据
  // 
  //----
  public struct HandleParam
  {
    short run;
    short turn;
    short height;
    short tilt;
  };

  //----
  // @brief 控制参数
  // 
  //----
  public struct ControlParam
  {
    bool begin; // 开始运行
    bool stop; // 急停 打算做一个功能就是为了防止程序疯跑，再怎么说没有急停开关好用
    byte robotmode;
  };

  //----
  // @brief 主控参数
  // 
  //----
  public struct Master
  {
    HandleParam handle;
    ControlParam control;
  };

  //----
  // @brief 机器人状态参数，供主控查看
  // 
  //----
  public struct RobotState
  {
    byte deviceState; // 电机状态
    float v;
    float height;
    float pitch;
    float yaw;
    float roll;
  };
}